### PR TITLE
bye bye Common.Logging, hello LibLog...

### LINKED ISSUE #847: Deprecate Common.Logging usage in favour of our own logging API
Let's drop Common.Logging which is an extra dependency that has already caused a number of issues in terms of assembly versioning, binding redirects and what not.

I propose we just use LibLog (https://github.com/damianh/LibLog) which is a single class file that you just copy paste.

### PR DESCRIPTION
**Pull request by adamralph**
fixes #847

### CODE REVIEW

**Comment by glennblock** [comment]
Can't wait for this one to pass!

**Comment by adamralph** [comment]
Argghhh LibLog induces a mono compiler warning - and it's a source package.

Guess I'll just alter it. We'll just have to be careful when we upgrade to a newer version.

**Comment by adamralph** [comment]
I think the mono compiler is telling big lies...

**Comment by adamralph** [comment]
Hopefully that should do it. I raised https://github.com/damianh/LibLog/issues/46 but I doubt there's anything Damian can do about it.

**Comment by adamralph** [comment]
OK, that's fixed the compiler warning. Now Travis is simply acting the goat as usual...

**Comment by adamralph** [comment]
Hooray! Green at last!

I only had to hammer the rebuild button in Travis about 7 or 8 times...

**Comment by glennblock** [comment]
OMG 80 files. Was this manual or search and replace?

File: Settings.StyleCop [hunk 1]
```diff
@@ -6,6 +6,15 @@
       <Value>nullable</Value>
     </CollectionProperty>
   </GlobalSettings>
+  <Parsers>
+    <Parser ParserId="StyleCop.CSharp.CsParser">
```
Comment (glennblock) [inline Settings.StyleCop]: Why is this here?

File: src/ScriptCs.Core/App_Packages/LibLog.3.1/LibLog.cs [hunk 2]
```diff
@@ -0,0 +1,1941 @@
+//===============================================================================
```
Comment (glennblock) [inline src/ScriptCs.Core/App_Packages/LibLog.3.1/LibLog.cs]: 1900 lines of code in one file really? Is there no refactored version? We can have a logging folder and put individual files there.

**Comment by adamralph** [comment]
It was manual

File: Settings.StyleCop (hunk 1 above)
Comment (adamralph) [inline Settings.StyleCop]: This stops StyleCop from analysing the LibLog file.

**Comment by glennblock** [comment]
I bit my tongue and lip and pressed merge.

**Comment by jrusbatch** [comment]
:clap:

**Comment by glennblock** [comment]
Thanks for putting in what must have been an incredibly boring, but useful effort :+1:

**Comment by adamralph** [comment]
I quite enjoyed it :smile:. I'm a bit special like that

**Comment by ztone** [comment]
Nice work @adamralph :tada:

File: Settings.StyleCop (hunk 1 above)
Comment (damianh) [inline Settings.StyleCop]: In ur codez, pollutin' ur stylez

File: src/ScriptCs.Core/App_Packages/LibLog.3.1/LibLog.cs (hunk 2 above)
Comment (damianh) [inline src/ScriptCs.Core/App_Packages/LibLog.3.1/LibLog.cs]: I made a conscious decision to ship as a single file. Don't want to crap stuff all over peoples' csprojs. Also helps reduce chances of accidental editing / refactoring. If you _do_ edit it, an update to a newer version won't work (smoothly). In my experience of using source code packages, I've found life is easier this way.

While it's also developed as a single file but even if I split that out I'd still ship as a single file.

Comment (damianh) [inline src/ScriptCs.Core/App_Packages/LibLog.3.1/LibLog.cs]: Oh and TinyIoC does the same thing for similar reasons.

Comment (glennblock) [inline src/ScriptCs.Core/App_Packages/LibLog.3.1/LibLog.cs]: @damianh I was initially against the idea, but as I reflected on it more, and the benefits like you cited I warmed up to it. Common.Logging needed to go, so this is a great replacement.

Thanks!

**Comment by filipw** [comment]
https://github.com/damianh/LibLog/releases/tag/v4.0.0

**Comment by damianh** [comment]
I've delisted 4.0.0 pending this https://github.com/damianh/LibLog/issues/48 Would appreciate your input on this. Would like to re-publish 4 as quickly as possible.
